src/core/dev_api/openvino/core/rt_info/weightless_caching_attributes.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/core/core_visibility.hpp"
+#include "openvino/core/runtime_attribute.hpp"
+
+namespace ov {
+
+/**
+ * @brief Holds weightless caching attributes of a single constant.
+ *
+ * WeightlessCacheAttribute class represents runtime info attribute that holds
+ * the values of original size of the constant in bytes and the binary offset of the
+ * constant's data in the weights file used by the weightless caching mechanism. It's
+ * not copyable in case the data was changed (the original node was replaced by a new
+ * one produced during the tranformation pipeline) - in that case weightless caching
+ * can't be used for that constant.
+ */
+class OPENVINO_API WeightlessCacheAttribute : public RuntimeAttribute {
+public:
+    OPENVINO_RTTI("WeightlessCacheAttribute");
+
+    WeightlessCacheAttribute() = default;
+
+    WeightlessCacheAttribute(size_t original_size, size_t bin_offset)
+        : original_size(original_size),
+          bin_offset(bin_offset) {}
+
+    bool is_copyable() const override;
+
+    size_t original_size;
+    size_t bin_offset;
```

Comment:
Set default values for members or consider removing default constructor.

Reply:
Deleted default constructor.